### DISCUSSION TITLE
Powershell Fonts Not Working

### DISCUSSION THREAD

**Discussion by HarryWangATX** [nvim-tree/nvim-tree.lua#1364] [Q&A]
### Description

When opening up the tree, there are just squares in place of the icons.

### Neovim version

```text
NVIM v0.7.0
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

### Operating system and version

Windows 11

### nvim-tree version

Commit 909

### Minimal config

```lua
require("nvim-tree").setup({
  sort_by = "case_sensitive",
  view = {
    adaptive_size = true,
    mappings = {
      list = {
        { key = "u", action = "dir_up" },
      },
    },
  },
  renderer = {
    group_empty = true,
  },
  filters = {
    dotfiles = true,
  },
})
```

### Steps to reproduce

1. Open any nvim file on Windows Powershell
2. :NvimTreeToggle
3. Icons seen are squares instead of actual icons.

### Expected behavior

Icons should be actual icons, but are just colored squares.

### Actual behavior

![image](https://user-images.githubusercontent.com/47877587/174507084-80ec1b1d-8475-482d-a708-b22f509c4a06.png)

**Comment by alex-courtis** [comment]
You will need to use a [patched font](https://www.nerdfonts.com/). See `:help nvim-tree-introduction`.

**Comment by HarryWangATX** [comment]
Hi, sorry I am a little new to this. Do which font would I download and load onto windows powershell on the website?

**Comment by HarryWangATX** [comment]
Did some googling, currently I'm running an instance of Powershell on Windows Terminal. I have Cascadia Code Font loaded on Windows Terminal which is one of the patched fonts I think. Also, I have oh-my-posh loaded on PowerShell and those glyphs load corrected in Windows Terminal.

**Comment by alex-courtis** [comment]
I'm using [patched hack](https://sourcefoundry.org/hack/) that you can see at the top of the [readme](https://github.com/kyazdani42/nvim-tree.lua)

I'm afraid I cannot help you install fonts or configure your terminal. Perhaps a user with Windows might be able to help.

**Comment by timii** [comment]
Hi @HarryWangATX,

Windows user here. I don't know if you still have the problem with your font, but I had the same issue using PowerShell as well. What helped me, was downloading and installing a patched font from the [nerd-fonts repository](https://github.com/ryanoasis/nerd-fonts#patched-fonts). They provide a few different patched versions of popular fonts.

**Example**: I saw in a comment before that you use the Cascadia Code font. If you want to use the same font, you can go into the [patched-fonts folder](https://github.com/ryanoasis/nerd-fonts/tree/master/patched-fonts) in the nerd-fonts repo and download the fonts you want to use from the [CascadiaCode folder](https://github.com/ryanoasis/nerd-fonts/tree/master/patched-fonts/CascadiaCode). The installation process is the same as for any other font (right-clicking the downloaded file and pressing install). After you installed the font, you need to restart your terminal and afterwards can choose the new font (probably has a "NF" in the name for "nerd-fonts"). Then the icons should work as expected.

Hope this helps a bit, and reply under this answer if you have any further questions.
